lib/pos_mempool.go

```diff
@@ -716,56 +720,43 @@ func (mp *PosMempool) GetIterator() MempoolIterator {
 // the main mempool. The temp mempool will have the most up-to-date readOnlyLatestBlockView, Height, and globalParams. Any
 // transaction that fails to add to the temp mempool will be removed from the main mempool.
 func (mp *PosMempool) Refresh() error {
-	mp.Lock()
-	defer mp.Unlock()
-
+	mp.RLock()
 	if !mp.IsRunning() {
 		return nil
 	}
 
-	if err := mp.refreshNoLock(); err != nil {
-		return errors.Wrapf(err, "PosMempool.Refresh: Problem refreshing mempool")
-	}
-	return nil
-}
+	validationView := mp.readOnlyLatestBlockView
+	mempoolTxns := mp.getTransactionsNoLock()
+	mp.RUnlock()
 
-func (mp *PosMempool) refreshNoLock() error {
-	// Create the temporary in-memory mempool with the most up-to-date readOnlyLatestBlockView, Height, and globalParams.
-	tempPool := NewPosMempool()
-	err := tempPool.Init(
-		mp.params,
-		mp.globalParams,
-		mp.readOnlyLatestBlockView,
-		mp.latestBlockHeight,
-		"",
-		true,
-		mp.maxMempoolPosSizeBytes,
-		mp.mempoolBackupIntervalMillis,
-		mp.feeEstimator.numMempoolBlocks,
-		mp.feeEstimator.cachedBlocks,
-		mp.feeEstimator.numPastBlocks,
-		mp.augmentedBlockViewRefreshIntervalMillis,
-	)
+	var txns []*MsgDeSoTxn
+	var txHashes []*BlockHash
+	for _, txn := range mempoolTxns {
+		txns = append(txns, txn.Tx)
+		txHashes = append(txHashes, txn.Hash)
+	}
+	copyValidationView, err := validationView.CopyUtxoView()
```

Comment:
Tests are failing because the validationView is nil. When would the validation view be nil? We should have a check in here somewhere that the view is nil or patch where it allows the view to be nil.

Reply:
Good observation. Added the check. I'll also fix the tests in a later PR.